effect/shared/src/main/scala/com/monovore/decline/effect/CommandIOApp.scala

```diff
@@ -0,0 +1,42 @@
+package com.monovore.decline.effect
+
+import cats.effect.{IO, IOApp, ExitCode}
+import cats.implicits._
+
+import com.monovore.decline._
+
+abstract class CommandIOApp(name: String, header: String, helpFlag: Boolean = true, version: String = "") extends IOApp {
+
+  def commandOpts: Opts[IO[ExitCode]]
+
+  private[this] val command: Command[IO[ExitCode]] = {
+    val showVersion = {
+      if (version.isEmpty) Opts.never
+      else {
+        val flag = Opts.flag(
+          long = "version",
+          short = "v",
+          help = "Print the version number and exit.",
+          visibility = Visibility.Partial
+        )
+        flag.as(IO(System.out.println(version)).as(ExitCode.Success))
+      }
+    }
+
+    Command(name, header, helpFlag)(showVersion orElse commandOpts)
+  }
+
+  override final def run(args: List[String]): IO[ExitCode] = {
+    def printHelp(help: Help): IO[ExitCode] =
+      IO(System.err.println(help)).as {
+        if (help.errors.size > 0) ExitCode.Error
+        else ExitCode.Success
+      }
+
+    for {
```

Comment:
When ambient args are set we totally ignore args here? That seems strange to me. Can we comment why?

Reply:
Yes, that line will ignore `args` when `ambientArgs` is not a `None`. If I am correct, `ambientArgs` should be a `Some` only when running in a NodeJS process, which made me implement the test only in the JVM because I was not using real command line arguments.

This is just a replication of the same behaviour in `core` for the `CommandApp`. It could also be safe to assume that, since `IOApp` is already cross-platform, we don't need the `ambientArgs` trick anymore.

Shall I get rid of them and rely on `args` only?

Reply:
Yep, you understand the idea right.

I actually don't know if `IOApp` does the right thing here --worth testing! (I can't see code that would do it, but I might be missing out.) If testing is too painful, we can probably skip -- this code at least shouldn't do any harm.

(If `IOApp` is not doing the right thing, that might be a nice upstream contribution, but we'd want to hang onto it in the meantime.)

Reply:
According to ScalaJS docs, there should be no need to do any tricks to get the command line arguments when running a ScalaJS process (their example HelloWorld takes a `def main(args: Array[String]): Unit`). But I keener of keeping the same code that is right now in `core` and potentially fix this in a PR that targets both modules.

Reply:
Yeah, historically they would just call `main` with an empty array. (Sketchy!) Maybe they've fixed it, but I'd want to test to be sure.

+1 to keeping the existing pattern and following up.